### ISSUE TITLE
Extreme slowdowns when installing multiple mods back-to-back

### ISSUE THREAD

**Issue by cprin21** [ralgar/proton-mo2-installer#1]
A common error that keeps coming up in the log for Mod Organizer 2's UI with both Rockerbacon's and this MO2 installer versions is the following:
EnableNonClientDpiScaling() failed for HWND 0x[this part changes] (120) (Call not implemented.)
Enough of that error coming up and the client begins to become unresponsive and requires a reboot of MO2.
![Screenshot_20220609_211719](https://user-images.githubusercontent.com/107228497/172971000-f46f8c03-a8fa-4a66-9659-84f4dfeef99e.png)

**Comment by ralgar** [comment]
The error message is a WINE/Proton issue, an [unimplemented syscall](https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-enablenonclientdpiscaling). However, I don't think this is the cause of the slowdown you're experiencing. In my experience, this is basically just log spam, and the only real side effect that I've noticed is that it causes the MO2 application to run a mini crash dump on exit. There is an option in MO2 to disable crash dumps entirely, but it doesn't actually seem to do anything - most likely an upstream MO2 issue.

In terms of UX, this isn't a serious issue, and there's nothing I can do about it anyway.

---

### Now for the actual problem:
**Are you using the NXM download links from your browser?**
I've experienced the same slowdown when using that functionality, and I *believe* it is down to how `protontricks` is calling the NXM handler executable. My best guess is that it's somehow breaking the Proton runtime.

I have an idea about how to fix this, which would involve adding IPC functionality to my `proton-shunt` utility. This would allow the xdg-mime handler to notify the shunt of a new download, which would then simply create a new process for the NXM handler. This should be a much more Proton-friendly way of doing things, and will hopefully resolve the issue. I'm going to experiment with it a bit this weekend, and hopefully have a working fix soon.

For now, the only solution is to simply restart MO2 every so often (or manually download and add mods, but that's a pain). It doesn't appear to affect gameplay in any way, as long as you restart MO2 before launching the game.

**Comment by cprin21** [comment]
> **Are you using the NXM download links from your browser?** I've experienced the same slowdown when using that functionality, and I _believe_ it is down to how `protontricks` is calling the NXM handler executable. My best guess is that it's somehow breaking the Proton runtime.

No, NXM download links do not work for me, logged into the same nexus account on my browser and in MO2 and everything, using RC5 of your proton-mo2-installer. I download everything manually and install by pointing to the mod's archive in my files.
Thanks for saying it was a good first issue, had to make a Github acct to make feedback because of how cool of a project this is (given that the original seems to have stopped being developed in January though I can't confirm that, just assuming since it hasn't gotten an update since then)

**Comment by ralgar** [comment]
> Thanks for saying it was a good first issue, had to make a Github acct to make feedback because of how cool of a project this is 

You gave all of the relevant information to the best of your knowledge, and even a picture. That's pretty great for a first issue! I appreciate the effort, and you saying this is a cool project as well. Hopefully we can get these bugs ironed out here.

---

> No, NXM download links do not work for me, logged into the same nexus account on my browser and in MO2 and everything, using RC5 of your proton-mo2-installer.

Ok, so there's another bug then, I opened a separate issue for that. Check #2 for more info.

---

### As for this issue:
I played around with this over the weekend, and I was way off base with the whole `protontricks` / NXM link idea. Turns out the issue is some kind of regression in the official (and other, such as GE) versions of Proton, and it only happens when installing mods in MO2. Around 10-15 mods installed back-to-back results in a noticeable slowdown.

Unfortunately the last *official* Proton version that I could get to work correctly is Proton 5.0. This is a very old version, from before Valve started building Proton against the Steam Linux Runtime. I also tried various versions of [Proton-GE](https://github.com/GloriousEggroll/proton-ge-custom) and the problem still remained.

A working solution (thanks to Frostworx, the author of SteamTinkerLaunch) is to use a custom-rolled version of Proton-TKG. I'd like to try using a 7.x version, so I'm going to try building 7.6 today, and see if that works. If 7.6 doesn't work, then 6.17 is at least confirmed working, so I will build and upload that version ASAP.

This also solves the issue of MO2 always doing a mini crash dump on exit.

---

TL;DR: One way or another I'll have a working solution up soon, and please check #2 regarding the NXM download links.

**Comment by ralgar** [comment]
@cprin21 This has been fixed now thanks to a custom Proton version, check [Release Candidate 7](https://github.com/ralgar/proton-mo2-installer/releases/tag/v1.0.0-RC7) . Let me know if you run into any more issues.
